src/Form/PrepaidBalanceReportsDownloadForm.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_m10n\Form;
+
+use Apigee\Edge\Exception\ClientErrorException;
+use Drupal;
+use Drupal\apigee_m10n\ApigeeSdkControllerFactoryInterface;
+use Drupal\apigee_m10n\MonetizationInterface;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\user\UserInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a form to to download prepaid balances.
+ */
+class PrepaidBalanceReportsDownloadForm extends FormBase {
+
+  /**
+   * The Apigee Monetization base service.
+   *
+   * @var \Drupal\apigee_m10n\MonetizationInterface
+   */
+  protected $monetization;
+
+  /**
+   * The Apigee SDK controller factory.
+   *
+   * @var \Drupal\apigee_m10n\ApigeeSdkControllerFactoryInterface
+   */
+  protected $sdkControllerFactory;
+
+  /**
+   * The user from route.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * An array of supported currencies.
+   *
+   * @var array
+   */
+  protected $supportedCurrencies;
```

Comment:
(nitpick:) This and `$billingDocuments` are only used inside one `buildForm` so they don't need to be a class variables.

Reply:
Fixed.